src/test/java/teammates/test/cases/browsertests/AdminHomePageUiTest.java

```diff
@@ -83,15 +85,21 @@ private void testCreateInstructorAction() throws Exception {
         InstructorAttributes instructorInBackend = getInstructorWithRetry(demoCourseId, instructor.email);
         assertEquals("Instructor Details must have 3 columns", homePage.getMessageFromResultTable(1));
 
-        String encryptedKey = getKeyForInstructorWithRetry(demoCourseId, instructor.email);
-        // use AppUrl from Config because the join link takes its base URL from build.properties
-        String expectedjoinUrl = Config.getAppUrl(Const.ActionURIs.INSTRUCTOR_COURSE_JOIN)
-                                        .withRegistrationKey(encryptedKey)
-                                        .withInstructorInstitution(institute)
-                                        .toAbsoluteString();
         assertEquals("Instructor AHPUiT Instrúctör WithPlusInEmail has been successfully created " + Const.JOIN_LINK,
                 homePage.getMessageFromResultTable(2));
-        assertEquals(expectedjoinUrl, homePage.getJoinLink(homePage.getMessageFromResultTable(2)));
+
+        String encryptedKey = getKeyForInstructorWithRetry(demoCourseId, instructor.email);
+
+        String expectedjoinUrl = Config.getAppUrl(Const.ActionURIs.INSTRUCTOR_COURSE_JOIN)
+                                                        .withRegistrationKey(encryptedKey)
+                                                        .withInstructorInstitution(institute)
+                                                        .toAbsoluteString();
+        String expectedJoinLinkQuery = new URL(expectedjoinUrl).getQuery();
```

Comment:
Can you try `getFile()`? According to the documentation, it has a combination of `filename` and `query`, in this case, `Const.ActionURIs.INSTRUCTOR_COURSE_JOIN` + `queryString`.